cgo/lapack.go

```diff
@@ -871,6 +871,11 @@ func (impl Implementation) Dbdsqr(uplo blas.Uplo, n, ncvt, nru, ncc int, d, e, v
 func (impl Implementation) Dgebrd(m, n int, a []float64, lda int, d, e, tauQ, tauP, work []float64, lwork int) {
 	checkMatrix(m, n, a, lda)
 	minmn := min(m, n)
+	ws := max(m, n)
+	if lwork == -1 {
+		work[0] = float64(ws)
```

Comment:
`work` must have always length at least 1, the reference always stores the optimum length (we don't but we should). Therefore it is strange to return work[0] = 0 if both m and n are 0 even though the callers should not assume work[0] >= 1.